pandora/transport.py

```diff
@@ -46,7 +46,11 @@ def function(*args, **kwargs):
                     retries_left -= 1
                     return func(*args, **kwargs)
 
-                except exceptions:
+                except exceptions as e:
+                    # Don't retry for PandoraExceptions - unlikely that result
+                    # will change for same set of input parameters.
+                    if isinstance(e, PandoraException):
```

Comment:
That sounds like a neat approach. Just a heads up that I am off the grid until Saturday in case I don't respond.